scene/gui/color_picker.cpp

```diff
@@ -422,14 +416,39 @@ ColorPicker::PickerShapeType ColorPicker::get_picker_shape() const {
 	return picker_type;
 }
 
+void ColorPicker::_add_preset_button(int p_size, const Color &p_color) {
+	ColorPresetButton *btn_preset = memnew(ColorPresetButton(p_color));
+	btn_preset->set_preset_color(p_color);
+	btn_preset->set_custom_minimum_size(Size2(p_size, p_size));
+	btn_preset->connect("gui_input", callable_mp(this, &ColorPicker::_preset_input), varray(p_color));
+	btn_preset->set_tooltip(vformat(RTR("Color: #%s\nLMB: Apply color\nRMB: Remove preset"), p_color.to_html(p_color.a < 1)));
+	preset_container->add_child(btn_preset);
+}
+
 void ColorPicker::add_preset(const Color &p_color) {
 	if (presets.find(p_color)) {
 		presets.move_to_back(presets.find(p_color));
+
+		// Find button to move to the end.
+		ColorPresetButton *btn_to_move = nullptr;
+		for (int i = 1; i < preset_container->get_child_count(); i++) {
+			ColorPresetButton *current_btn = Object::cast_to<ColorPresetButton>(preset_container->get_child(i));
+			if (current_btn && p_color == current_btn->get_preset_color()) {
+				btn_to_move = current_btn;
+				break;
+			}
+		}
+		if (btn_to_move) {
+			preset_container->move_child(btn_to_move, preset_container->get_child_count() - 1);
+		}
 	} else {
 		presets.push_back(p_color);
 		preset_cache.push_back(p_color);
+
+		int hsep = preset_container->get_theme_constant(SNAME("hseparation"));
+		int preset_size = (int(get_size().width) - (hsep * (preset_column_count - 1))) / preset_column_count;
```

Comment:
This snippet appears twice in the code. You could make a (inline?) function to avoid code duplication, e.g.
```
inline int _get_preset_size() { return (int(get_size().width) - (preset_container->get_theme_constant(SNAME("hseparation")) * (preset_column_count - 1))) / preset_column_count; }
```

Reply:
Done.